### PR TITLE
Begin using Terraform Workspaces and add backend setup configuration

### PR DESCRIPTION
**Pull request by nickclyde**
Resolves the following tickets (sorry for cramming these all into a single PR, but these changes would be hard to do piece by piece):

- [Start using Terraform Workspaces](https://reportstream-service-request.clickup.com/t/3b9ce69)
- [Bootstrap initial Terraform setup by splitting into "setup" and "implementation" folders](https://reportstream-service-request.clickup.com/t/3b9cedx)
- [Add GitHub workflow for initial tfstate bucket setup](https://reportstream-service-request.clickup.com/t/3c0b3zu)
- [Add a `terraform validate` CI check](https://reportstream-service-request.clickup.com/t/3c0b39g)
- [Write a script to automate setup of Workload Identity Federation](https://reportstream-service-request.clickup.com/t/3c0b1fn)
- [Write a "Quick Start" guide in README.md](https://reportstream-service-request.clickup.com/t/3c0b2va)

This is a pretty big reorganization of our Terraform code. Instead of having different directories in `vars` for each environment, we can now instead use a pretty cool feature of Terraform called [Workspaces](https://www.terraform.io/cli/workspaces) to isolate state by environments on the CLI. More info about how to use Workspaces is included in the changes to `docs/getting_started.md`.

This also adds a `terraform/setup` folder which sets up a bucket for storing tfstate separately from the main config, which now lives in `terraform/implementation`. Running `terraform apply` from the setup folder will spit out a bucket name, which you will then pass in when running `terraform apply` in the implementation folder.

There is also a new GitHub workflow called `terraformSetup` which will run this initial bootstrap step to create the state bucket. The newly named `deployment` step knows how to construct the name of the bucket automatically and pass it to terraform as backend config.

I also cleaned up some of our config to remove redundant settings that are using the defaults, and added the terraform workspace name and project ID to bucket names in place of random strings.

I added another CI check which runs `terraform validate` against our terraform configs, which goes beyond `terraform fmt` and "runs checks that verify whether a configuration is syntactically valid and internally consistent, regardless of any provided variables or existing state. It is thus primarily useful for general verification of reusable modules, including correctness of attribute names and value types".

I also added a script `quick-start.sh` which automates some of the initial setup end users will need to do after forking this repo, including creating service accounts and setting up Workload Identity Federation [as outlined here](https://github.com/google-github-actions/auth#setup). The script runs all the necessary `gcloud` commands and spits out the secrets which should be loaded into the repository secrets at the end.

Finally, I added a new section to the README called `Quick Start` which outlines the process of forking the repo, running the script, and triggering GitHub Actions to deploy a full pipeline.

Whew, sorry for the wall of text and code. Just having one of those days where my fingers fly :sweat_smile:

### CODE REVIEW

**Comment by celestemespinoza** [comment]
Task linked: [CU-3b9ce69 Start using Terraform Workspaces](https://app.clickup.com/t/3b9ce69)

**Comment by celestemespinoza** [comment]
Task linked: [CU-3b9cedx Bootstrap initial Terraform setup by splitting into "setup" and "implementation" folders](https://app.clickup.com/t/3b9cedx)

**Comment by celestemespinoza** [comment]
Task linked: [CU-3c0b1fn Write a script to automate setup of Workload Identity Federation](https://app.clickup.com/t/3c0b1fn)

**Comment by celestemespinoza** [comment]
Task linked: [CU-3c0b2va Write a "Quick Start" guide in README.md](https://app.clickup.com/t/3c0b2va)

**Comment by celestemespinoza** [comment]
Task linked: [CU-3c0b39g Add a `terraform validate` CI check](https://app.clickup.com/t/3c0b39g)

**Comment by celestemespinoza** [comment]
Task linked: [CU-3c0b3zu Add GitHub workflow for initial tfstate bucket setup](https://app.clickup.com/t/3c0b3zu)

File: docs/README.md [hunk 1]
```diff
@@ -24,6 +25,34 @@
...
+### Quick Start
+
+To deploy this pipeline to your own Google Cloud environment, follow these steps.
+  
+  Be sure to replace all instances of `CDCgov` in GitHub URLs with your user or organization name.
+  1. [Install the gcloud CLI](https://cloud.google.com/sdk/docs/install-sdk)
+  1. [Fork this repository](https://github.com/CDCgov/phdi-google-cloud/fork) into your personal or organization account
+  1. Clone your newly forked repository to your local machine by running:
+
+         git clone https://github.com/CDCgov/phdi-google-cloud.git
```
Comment (DanPaseltiner) [inline docs/README.md:37]: Doesn't this clone our repo and not their fork of it?

File: quick-start.sh [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+if [ -z "$PROJECT_ID" ]; then
+    echo "Error: Project ID not found. To list projects, run 'gcloud projects list'."
+    exit 1
+fi
+
+# Create a service account
+gcloud iam service-accounts create "github" \
+  --project "${PROJECT_ID}" \
+  --display-name "github"
+
```
Comment (DanPaseltiner) [inline quick-start.sh:45]: What level of permissions does a user need to create service accounts with sufficient access for deploying Terraform. Should we document this as a requirement for running this script?

File: quick-start.sh [hunk 3]
```diff
@@ -0,0 +1,106 @@
+#!/bin/bash
```
Comment (DanPaseltiner) [inline quick-start.sh:1]: This is awesome! Thanks for putting it together.

File: docs/README.md (hunk 1 above)
Comment (nickclyde) [reply docs/README.md:37]: Yeah, but on line 32 I wrote "Be sure to replace all instances of `CDCgov` in GitHub URLs with your user or organization name." That might be easy to miss, though. I couldn't think of a better way to do it, besides maybe replacing `CDCgov` with `myuser` or something like that?

File: .github/workflows/deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,64 @@
+name: Deployment
```
Comment (DanPaseltiner) [inline .github/workflows/deployment.yaml:1]: I'm a little confused about how we handle multiple environments and deployment logic between them. Would it be worth seeing if we can get a second GCP project so we can make our own `dev` and `prod` environments and work out the associated CD logic? This could totally be another PR.

**Review by DanPaseltiner — CHANGES_REQUESTED** [review]
This looks absolutely amazing. Couple little things to fix and I like to briefly discuss handling multiple environments before merging. Thanks for all your work on this!

File: docs/README.md (hunk 1 above)
Comment (DanPaseltiner) [reply docs/README.md:37]: Given that `git clone https://github.com/CDCgov/phdi-google-cloud.git` will run I think this creates an easy gotcha mistake for new users.

File: terraform/modules/fhir/main.tf [hunk 5]
```diff
@@ -1,11 +1,11 @@
 resource "google_healthcare_dataset" "dataset" {
-  name      = "${var.dataset_name}-${time_static.current_date.unix}"
+  name      = "phdi-${terraform.workspace}-dataset"
```
Comment (nickclyde) [inline terraform/modules/fhir/main.tf:2]: needs to be unique

**Review by DanPaseltiner — APPROVED** [review]
Thanks for talking things through. This looks great!
